packages/components/src/tools-panel/stories/index.js

```diff
@@ -79,7 +77,97 @@ export const _default = () => {
 	);
 };
 
+const { Fill: ToolsPanelItems, Slot } = createSlotFill( 'ToolsPanelSlot' );
+const panelId = 'unique-tools-panel-id';
+
+export const WithSlotFillItems = () => {
+	const [ attributes, setAttributes ] = useState( {} );
+	const { width, height } = attributes;
+
+	const resetAll = ( resetFilters = [] ) => {
+		let newAttributes = {};
+
+		resetFilters.forEach( ( resetFilter ) => {
+			newAttributes = {
+				...newAttributes,
+				...resetFilter( newAttributes ),
+			};
+		} );
+
+		setAttributes( newAttributes );
+	};
+
+	const updateAttribute = ( name, value ) => {
+		setAttributes( {
+			...attributes,
+			[ name ]: value,
+		} );
+	};
+
+	return (
+		<SlotFillProvider>
+			<ToolsPanelItems>
```

Comment:
This example and the documentation efforts made me think whether `panelId` should be part of the component. Should we create an opinionated Fill and Slot and keep them in the block editor instead?

The difference would be that `ToolsPanelItem` from the `@wordpress/block-editor` package would be fill that handles all the logic related to `panelId` and renders `ToolsPanelItem`. I don't know if I'm doing a good job explaining it. 

So the goal would be to achieve the following:

```js
const { Fill, Slot } = createSlotFill( 'ToolsPanelSlot' );

function shouldRender( panelId ) {
    // checks somehow whether to render the panel based on the value passed from slot 
}

const ToolsPanelItemFill = ( { panelId, ...props } ) => {
    return (
        <Fill>
            { shouldRender( panelId ) && <ToolsPanelItem { ...props } /> }
        </Fill>
    );
};

const ToolsPanelSlot = ( { panelId, ...props } ) => {
    return (
        <ToolsPanel { ...props }>
            <Slot fillProps={ panelId }></Slot>
        </ToolsPanel>
    );
}; 
```

Reply:
That's an interesting idea. I don't think I fully understand the implications yet but will explore this tomorrow.

It makes sense in terms of being able to remove the need for `panelId` on the `ToolsPanel` itself. At this stage though, I don't have much of an idea how the individual `ToolsPanelItem`s would be working.

Reply:
You probably can play with the Storybook story first and see if you can make it work this way. As noted in another place, given that both updated components are marked as experimental don't treat my comment as a blocker.